### PR TITLE
[l2-deversifi] L2-deversifi strategy

### PR DESCRIPTION
**Pull request by nicob21**
Implementation of l2-deversifi strategy

### CODE REVIEW

File: src/strategies/l2-deversifi/index.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        }
+      })
+    );
+  });
+
+  const results = await Promise.all(promises);
+  const resultsJson = await Promise.all(results.map((r: any) => r.json()));
+
+  const data: any = resultsJson.reduce(
+    (res: any, item: any) => [...res, ...item.score],
```
Comment (ChaituVR) [inline src/strategies/l2-deversifi/index.ts:42]: There is a error here, which is failing the test cases, would be great if you can fix it
```TypeError: item.score is not iterable

      28 |     const results = await Promise.all(promises);
      29 |     const resultsJson = await Promise.all(results.map((r) => r.json()));
    > 30 |     const data = resultsJson.reduce((res, item) => [...res, ...item.score], []);
```

File: src/strategies/l2-deversifi/index.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  const { api, token, limit = 300 } = options;
+
+  const pages = Math.ceil(addresses.length / limit);
+  const promises: any = [];
+
+  let api_url = api;
+  api_url += `?blockNumber=${snapshot}`;
+  api_url += `&token=${token}`;
+
+  Array.from(Array(pages)).forEach((x, i) => {
```
Comment (ChaituVR) [inline src/strategies/l2-deversifi/index.ts:25]: Looks like here you are trying to make a API fetch call for each address, which won't scale for proposals with more addresses, is there any other approach you can try? Where you can get data in a single request

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: src/strategies/l2-deversifi/index.ts (hunk 2 above)
Comment (plutoegg) [reply src/strategies/l2-deversifi/index.ts:25]: This is not per address, but per 300 address (so if its more than 300 addresses requested it splits it into multiple requests).

Comment (ChaituVR) [reply src/strategies/l2-deversifi/index.ts:25]: @plutoegg Cool then :) just noticed that, Thanks for letting me know

**Review by ChaituVR — CHANGES_REQUESTED** [review]
@plutoegg please resolve conflicts :)

**Comment by plutoegg** [comment]
> @plutoegg please resolve conflicts :)

Done!

**Review by ChaituVR — COMMENTED** [review]
LGTM

**Comment by plutoegg** [comment]
Does another snapshot team member need to merge? We are keen to test it out with the UI :)

**Review by bonustrack — APPROVED** [review]

**Review by ChaituVR — APPROVED** [review]

**Comment by bonustrack** [comment]
@plutoegg Thanks for your patience will get the strategy up in a moment

**Comment by ChaituVR** [comment]
@plutoegg Strategy is live here: https://snapshot.org/#/strategy/l2-deversifi Thanks :)

**Pull request merged**